File: output/ChatGPT/avatar/Java/C++/codeforces_147_A.cpp
#include <iostream>
#include <string>

using namespace std;

bool isLatinLetter(char c) {
    return c >= 'a' && c <= 'z';
}

bool isPunctuation(char c) {
    if (c == '.') return true;
    if (c == ',') return true;
    if (c == '!') return true;
    if (c == '?') return true;
    return false;
}

int main() {
    string s;
    getline(cin, s);
    string result;
    result += s[0];
    int n = s.length();
    for (int i = 1; i < n; i++) {
        char c = s[i];
        if (isLatinLetter(c)) {
            if (!isLatinLetter(s[i - 1])) {
                result += ' ';
            }
            result += c;
        } else if (isPunctuation(c)) {
            result += c;
        }
    }
    cout << result << endl;
}

File: output/ChatGPT/avatar/Java/Python/codeforces_147_A.py
import sys

def is_latin_letter(c):
    return c >= 'a' and c <= 'z'

def is_punctuation(c):
    if c == '.':
        return True
    if c == ',':
        return True
    if c == '!':
        return True
    if c == '?':
        return True
    return False

s = sys.stdin.readline().strip()
sb = []
n = len(s)
sb.append(s[0])
for i in range(1, n):
    c = s[i]
    if is_latin_letter(c):
        if not is_latin_letter(s[i - 1]):
            sb.append(' ')
        sb.append(c)
    elif is_punctuation(c):
        sb.append(c)

print(''.join(sb))